lib/plugins/aws/lib/naming.js

```diff
@@ -545,8 +545,17 @@ module.exports = {
   getLambdaAtEdgeInvokePermissionLogicalId(functionName) {
     return `${this.getLambdaLogicalId(functionName)}InvokePermission`;
   },
-  getCloudFrontOriginId(functionName, path) {
-    let originId = this.getNormalizedFunctionName(functionName);
+  getCloudFrontOriginId(originObj) {
+    const isS3Origin = originObj.S3OriginConfig !== undefined;
```

Comment:
Let's do `Boolean(originObj.S3OriginConfig)` (we also should discard values as `null` or `false`)